### DISCUSSION TITLE
Support for VisionOS

### DISCUSSION THREAD

**Discussion by llfab** [StereoKit/StereoKit#845] [General]
I have used StereoKit on Hololens and I liked the experience very much compared to cumbersome Unity. Is it completely unthinkable to support VisionOS and Apple Vision Pro in the future?

**Comment by maluoi** [comment]
It's not completely unthinkable, but it is a _very_ large chunk of work! Supporting WebXR is far likelier to happen, which will at least get you onto visionOS.

Major work required to do add native visionOS support would be:
- A Metal rendering backend
- A Mac/visionOS platform backend
- An ARCore XR backend

This is basically a full stack of engine code, all of which is only useful on visionOS.

And then this would only get you immersive apps on visionOS! Unity's Polyspatial for side-by-side apps is basically just ripping out Unity's scene-graph representation and attaching that to Apple's rendering system. Not exactly an idea that makes much sense for StereoKit.

If there are people in the community interested in doing some of this work, I'd be pretty excited about that! But I'm not much of a Mac dev myself, and this isn't likely to bubble up to the top of my own priority list anytime soon.

**Reply by llfab** [reply]
Thank you very much for these insights. Yes, only after having raised this question I nonticed that (of course) Vision Pro is not based at all on OpenXR. I appreciate very much what you have done so far. I wonder why there is not a lot more like StereoKit as an abstraction of OpenXR. 

We are working heavily with https://github.com/AvaloniaUI/Avalonia who are also aiming towards platforms like VisionOS. In my perfect world I see a collaboration of StereoKit and Avalonia.
